src/resource/archive/ArchiveManager.h

```diff
@@ -38,6 +38,7 @@ class ArchiveManager {
     std::shared_ptr<std::vector<std::string>> ListFiles(const std::string& searchMask = "");
     std::shared_ptr<std::vector<std::string>> ListFiles(const std::list<std::string>& includes,
                                                         const std::list<std::string>& excludes);
+    std::shared_ptr<std::vector<std::string>> ListDirectories(const std::string& pattern);
```

Comment:
```suggestion
    std::shared_ptr<std::vector<std::string>> ListDirectories(const std::string& pattern = "");
```

i figure we want to support someone calling `ListDirectories()` without a param

Reply:
also the name "pattern" feels a little ambiguous. for `ListFiles` we use `searchMask` which seems like a good name here too